src/main/java/com/google/sps/tool/WordSearch.java

```diff
@@ -0,0 +1,104 @@
+package com.google.sps.tool;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+import java.util.Collections;
+import java.util.Random;
+import java.lang.Math;
+
+public class WordSearch {
+  private static HashMap<Character, ArrayList<String>> alphabetWordMap =
+      new HashMap<Character, ArrayList<String>>();
+
+  public static ArrayList<String> generateWordOptions(String correctAnswer) {
+    ArrayList<String> quizWords = new ArrayList<>();
+    quizWords.add(correctAnswer);
+
+    if (!alphabetWordMap.containsKey('a') && !initMap()) {
+      // If the dictionary file can't be read then it just return the correct Answer
+      return quizWords;
+    }
+
+    char firstLetterOfAnswer = correctAnswer.charAt(0);
+    ArrayList<String> wordList = alphabetWordMap.get(Character.toLowerCase(firstLetterOfAnswer));
+    Boolean wordFound = false;
+
+    for (int i = 0; i < wordList.size(); i++) {
+      if (wordList.get(i).equals(correctAnswer)) {
+        wordFound = true;
+        getRandomWords(i, wordList, quizWords);
+        break;
+      }
+    }
+
+    // If exact word could not be found, then generate random words with the same first letter
+    if (!wordFound) {
+      Random random = new Random();
+      for (int i = 0; i < 3; i++) {
+        int index = random.nextInt(wordList.size());
+        if (!quizWords.contains(wordList.get(index))) {
+          quizWords.add(wordList.get(index));
+        } else {
+          i--;
+        }
```

Comment:
```java
      for (int i = 0; i < 3;) {
        int index = random.nextInt(wordList.size());
        if (!quizWords.contains(wordList.get(index))) {
          quizWords.add(wordList.get(index));
          ++i;
        }
      }
```

Reply:
I would even write this as a while loop — when I see a for loop I assume it will take a certain number of iterations, but when I see a while loop I know that the number of iterations is uncertain.

Reply:
done.